Windows: Support named pipe hardening

olljanat opened the pull request:
In production plugins only NT AUTHORITY\SYSTEM which run Docker engine should have access plugin named pipes.

olljanat committed:
Windows: Support named pipe hardening

In production plugins only NT AUTHORITY\SYSTEM which run Docker engine
should have access plugin named pipes.

Signed-off-by: Olli Janatuinen <olli.janatuinen@gmail.com>

akerouanton reviewed (approved):
SGTM

akerouanton commented:
Dumb question:

> Those named pipes cannot be overwritten which why they also need to be closed properly when plugin process exit.

What happens if the plugin crashes?

olljanat commented:
> Those named pipes cannot be overwritten which why they also need to be closed properly when plugin process exit.

This seems to be invalid comments which I had forgot to there. On some earlier draft I had added `defer l.Close()` which I though that was needed by later realized that it will break connectivity from Docker.

Removed it now also from commit message.

> What happens if the plugin crashes?

It closes normally. User mode services cannot leave named pipes behind when they close/crash.
